test/blackbox/common/BlackboxTestsLivelinessQos.cpp

```diff
@@ -2039,6 +2040,175 @@ TEST(LivelinessTests, correct_liveliness_state_one_writer_multiple_readers)
     ASSERT_EQ(reader.sub_wait_liveliness_lost_for(2, std::chrono::seconds(4)), 2u);
 }
 
+/**
+ * This is a regression test for redmine issue #21189.
+ *
+ * The test ensures that liveliness changed status is not affected by writers on a topic different from
+ * the one of the reader.
+ *
+ * The test creates two readers and two writers, each reader and writer pair on a different topic.
+ * Writing a sample on one writer should not affect the liveliness changed status of the other reader.
+ * Destroying the writer should not affect the liveliness changed status of the other reader.
+ */
+static void test_liveliness_qos_independent_topics(
+        const std::string& topic_name,
+        eprosima::fastdds::dds::ReliabilityQosPolicyKind reliability_kind)
+{
+    const auto lease_dutation_time = std::chrono::seconds(1);
+    const eprosima::fastdds::Duration_t lease_duration(1, 0);
+    const eprosima::fastdds::Duration_t announcement_period(0, 500000000);
```

Comment:
Otherwise timing on execution might cause a false negative
```suggestion
    const eprosima::fastdds::Duration_t announcement_period(0, 250000000);
```